packages/sku/src/types/types.ts

```diff
@@ -226,6 +227,22 @@ export interface SkuConfig {
    */
   dangerouslySetWebpackConfig?: (skuWebpackConfig: any) => any;
 
+  /**
+   * This function provides a way to modify sku's Vite configuration.
+   * It should only be used in exceptional circumstances where a solution cannot be achieved by adjusting standard configuration options.
+   *
+   * Before customizing your Vite configuration, please reach out in [#sku-support](https://seek.enterprise.slack.com/archives/CDL5VP5NU) to discuss your requirements and potential alternative solutions.
+   *
+   * As sku creates two Vite configs (`client` & `server|render`), this function will actually run twice.
+   * If you only need to modify one of these configs, then you can check `config.mode` within.
```

Comment:
I'm not sure if this is correct. `mode` in the config is used to override the configured `--mode` flag. It's also typically either `production` or `development`, rather than `client|server|render`. As mentioned in my other comment, if the full `config` hook signature is given, that would enable declaring specific config for a specific build.